### ISSUE TITLE
Use CHT reference app as default

### ISSUE THREAD

**Issue by garethbowen** [medic/medic#5914]
When starting for the first time we should load all the configuration, icons, forms, and so on to allow the first run experience to be gentler.

The easiest way to do this is to have API detect a first run and call medic-conf on startup however this means shipping a copy of medic-conf just for this which I'm not comfortable with.

Another option is to use medic-conf to turn the config into a collection of docs which can be attached to the published ddoc and merely uploaded on first start.

Finally you can delete the `config.default.json` from api.

Eventually we want to have e2e tests running against the reference app but for now just make sure they keep working and we'll migrate them over time.

Don't start on this until the CHT reference app is merged to this repo.

**Comment by kennsippell** [comment]
Instead of making this change in API (and adding the whole medic-conf dependency there), could we just detect first-run and deploy the CHT config 1) via grunt for local dev machines, and 2) as part of our process for building docker images?

**Comment by garethbowen** [comment]
Horti v2 doesn't fit with this timeline so it's not an option. I _think_ what you're suggesting is covered by my preferred option listed above:

> Another option is to use medic-conf to turn the config into a collection of docs which can be attached to the published ddoc and merely uploaded on first start.

That way we ship just a bunch of docs so API doesn't need the medic-conf dependency, it just uploads the docs on first run.

**Comment by kennsippell** [comment]
Plan sounds reasonable.

Also spoke with Gareth offline. For the scenario of deploying vPrevious (eg. v3.2.0) with "default config" and upgrading to 3.7 via Horti - this upgraded version will *not* received the new default config (CHT-Ref-App).

**Comment by kennsippell** [comment]
> Don't start on this until the CHT reference app is merged to this repo.

I'll start with the required medic-conf change, which I believe we can safely do in parallel while waiting for the CHT Reference App to merge into the repo.

**Comment by kennsippell** [comment]
Do we also want a default dataset as part of this?

**Comment by garethbowen** [comment]
We certainly don't want to load a default data set automatically because it's hard to bulk delete everything out. I can see us distributing a "demo" data set that's easy to import but that's a job for another day.

**Comment by abbyad** [comment]
We are including a very small demo data set in the reference app config folder that (optionally) can be pushed with `medic-conf` to help with getting oriented in the app.

**Comment by kennsippell** [comment]
Hey @newtewt and @ngaruko! This issue is the critical path for the 3.7 release. 

It consists of two parts - one in webapp and one in medic-conf. The medic-conf change is ready for review on `5914-archive` https://github.com/medic/medic-conf/pull/232.

**Comment by kennsippell** [comment]
Webapp change ready for AT in `5914-cht-default`

**Comment by kennsippell** [comment]
@ngaruko This is safe to merge if it passes AT.

**Comment by ngaruko** [comment]
@kennsippell 
In `permissions`, I see a few new roles such as 
![image](https://user-images.githubusercontent.com/6979995/65847170-e9e4cc80-e39c-11e9-87a9-3b6cc31cc43d.png)
```"program_officer",
            "district_admin", "chw_supervisor",
            "chw"
```
Should we have those roles defined in `roles`:
![image](https://user-images.githubusercontent.com/6979995/65847231-257f9680-e39d-11e9-8cfd-2956f489c058.png)

**Comment by kennsippell** [comment]
Hey @ngaruko! This configuration won't deploy if upgrade a docker container running v3.2.1/v.3.6.1 to this branch (`5914-cht-default`). In this case, a configuration is already present and the CHT will not be used as the default. 

To AT this, you'll have to build a docker image to directly deploy this branch, or AT this branch from a dev instance. Once you successfully trigger the default config to deploy, you should see the new roles appearing in the admin console. 

![Screenshot from 2019-09-30 00-30-58](https://user-images.githubusercontent.com/9014751/65857886-9f6e3c00-e319-11e9-8711-5b56bf879450.png)

**Comment by newtewt** [comment]
@ngaruko  If you clean up your docker env and then run this compose it will download the medic-os image with CHT as default that I've pushed to my repo. I'll also try and get on this today as well.

[clean up docker ](https://github.com/medic/medic-docs/blob/62786e406c38827bbb9b95a32c37de9627f3fe8e/installation/public-docker-image-setup.md#delete--re-install)

```
version: '3.7'

services:
  medic-os:
    container_name: medic-os
    image: newtewt/medic-cht-default
    volumes:
      - medic-data:/srv
    ports:
     - 80:80
     - 443:443
    working_dir: /srv
    depends_on:
      - haproxy
    networks:
      - medic-net
    environment:
      - DOCKER_NETWORK_NAME=haproxy
      - DOCKER_COUCHDB_ADMIN_PASSWORD=$DOCKER_COUCHDB_ADMIN_PASSWORD

  haproxy:
    container_name: haproxy
    image: medicmobile/haproxy:rc-1.16
    volumes:
      - medic-data:/srv    
    environment:
      - COUCHDB_HOST=medic-os
      - HA_PASSWORD=$DOCKER_COUCHDB_ADMIN_PASSWORD
    networks:
      - medic-net

volumes:
  medic-data:
    name: medic-data

networks:
  medic-net:
    name: medic-net

```

**Comment by newtewt** [comment]
I feel that the CHT is being deployed properly as the default config. All the forms, resources, 1 translations, settings, contact summary tasks and targets are showing. I'm going to spend some more time but likely will merge by end of day.

**Comment by garethbowen** [comment]
@newtewt Can this be closed now?

**Comment by newtewt** [comment]
yeah sorry I moved to done and it didn't close as I thought it should.
